### ISSUE TITLE
Missing BERTMGR32.xll

### ISSUE THREAD

**Issue by tranzmatt** [sdllc/Basic-Excel-R-Toolkit#31]
I had to reinstall BERT as some sort of conflict was happening between Excel Addons, but now something seems amiss.  BERTMGR32.xll is no longer being installed into the Roaming\BERT directory, so Excel is having fits about it when I fire it up.  I ran the msiexec with logging and while I see:

RegXLL:  setting OPEN entry: "C:\Users\DrMCC\AppData\Roaming\BERT\BERTMGR32.xll"

I never see any install entry like this:

InstallFiles: File: BERT32.xll,  Directory: C:\Users\DrMCC\AppData\Roaming\BERT\,  Size: 667648

So BERTMGR32.xll doesn't seem to be copied into the folder.  Can anyone verify this is the case?  I just tried it with a fresh download.  Same issue.  I'd built the project with my Visual Studio for kicks, but that can't be a reason for the failure, can it?

Matt

**Comment by alexsdllc** [comment]
We don't install BERTMGR any more -- it's still in the VS project, but the ribbon menu has replaced it for loading the other xll.  Are you using an old installer?  If so, what's the version?  I'm pretty sure we're not setting that registry entry in the current installer (although we might still be logging that message).

If you want to jump to current, I'd suggest cleaning up old files and then using the (current) installer.  You can switch to your build after the install.  To clean up, try the uninstaller, and if that doesn't work try deleting the BERT directory.

If Excel is complaining about a missing BERTMGR, and all you want is for it to stop complaining, you can do this via the UI: go to Excel options -> add-ins and try to select that add-in.  When it's not found Excel will offer to remove it from the list.

**Comment by tranzmatt** [comment]
It finally seemed to connect once I repeatedly un-installed BERT, scrubbed out everything in the registry and re-installed.  I don't know what got dorked up, but I'll keep that in mind.  The BERT directory in my Roaming folder was deleted by the installer and I had removed any examples/functions myself.  

I'm using 1.20, or which ever version is currently available for download.  I only built the VS project as an exercise to see the internals, never installed it.

Matt

**Comment by duncanwerner** [comment]
Heads up that functions now go under documents/BERT/functions (this should be easier for some users to locate), and it handles more than one file.  If you had not removed your functions before cleaning up, it still reads the functions file under roaming/BERT but it will give you a warning.

Regarding the "OPEN entry" -- we're not writing that registry entry, but the installer logic is still there to clean up from previous installs which did write it.  The log line should be fixed.  It's not in the installer, it's in a module (owing to a limitation in MSI).

**Comment by tranzmatt** [comment]
Yes, I saw the new location of the functions & examples folder.  I had cleaned it.  I think the conflict in my add-ins was the root cause.  Not sure how, exactly, but it seems to be working, so I'll take the win and move on.

**Comment by nksingh62** [comment]
I was working on BERT 1.24. During adding a button in BERT ribbon, program was crashed with message that there is serious problem in ribbon and suggested to use Add Ins  of trusted source. When I restarted Excel, Add-In tab was missing from menu. I tried with BERT version 1.24 and 1.33 after completely removing (by renaming BERT in .../document folder) but failed to recovering Add-In tab again.
I am using Excel 2007. After fresh installation of BERT, Excel Options shows Bertmgr32 in list but failed to locate it BERT folder under Roaming. In other computer where BERT is working successfully, I could not locate Bertmgr32.xll. Whether it is problem of locating Bertmgr32.xll or some thing else? Whether registry was marked, when Excel crashed, for reinstalling some components of BERT?

**Comment by alexsdllc** [comment]
Excel has disabled the ribbon menu.  Go to the add-ins dialog (Office menu > Options > Add-ins), select manage *Disabled Items* and click go.  Re-enable the bert ribbon menu.  If you have uninstalled, you might have to reinstall.  Don't worry about bertmgr32.xll, that's not related.

**Comment by nksingh62** [comment]
I did it. When I selected "Bert Ribbon Menu" from "Disabled application menu" and pressed "GO", only Bertmgr32 is showing in list of available Add-Ins which is not found in appropriate location. I tried to attach BERTRibbon2x86.dll  with "Bert Ribbon menu class" through automation button, available in "List of available Add-Ins" dialog box. It shows I do not have sufficient permission. I also deleted all information related to BERT and reinstalled it but failed to get "Add-In" tab in Excel menu.

**Comment by nksingh62** [comment]
I did it. When I selected "Bert Ribbon Menu" from "Disabled application menu" and pressed "GO", only Bertmgr32 is showing in list of available Add-Ins which is not found in appropriate location. I tried to attach BERTRibbon2x86.dll  with "Bert Ribbon menu class" through automation button, available in "List of available Add-Ins" dialog box. It shows that I do not have sufficient permission. 
I also deleted all information related to BERT. After deleting all information from registry, BERT is removed from list of available Add In application. It was not removing after simple "uninstallation" of BERT. However, "BERT Ribbon menu class" is not removed from available list of Automation Server even after deleting all information from registry.  I could not understand how my computer is memorizing that installation of BERT is not fresh and it was crashed earlier.

**Comment by nksingh62** [comment]
Problem is solved with help of issues discussed [here](https://ardamis.com/2010/02/26/fixing-word-2007-add-in-issues/). It may help to users of BERT if Excel crashed.

**Comment by gardark** [comment]
I had this problem. The solution for me was to go to Disabled Items í Add-ins.
